spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/AbstractDeployerSupport.java

```diff
@@ -60,8 +61,17 @@ final protected LocalDeployerProperties getLocalDeployerProperties() {
 	protected String[] buildJarExecutionCommand(String jarPath, AppDeploymentRequest request) {
 		ArrayList<String> commands = new ArrayList<String>();
 		commands.add(properties.getJavaCmd());
-		commands.add("-jar");
-		commands.add(jarPath);
+
+		Map<String, String> envProps = request.getEnvironmentProperties();
+		if (envProps.containsKey("main") && envProps.containsKey("classpath")) {
+			commands.add("-cp");
+			commands.add(envProps.get("classpath"));
+			commands.add(envProps.get("main"));
+		}
+		else {
+			commands.add("-jar");
+			commands.add(jarPath);
+		}
```

Comment:
Should we handle the case where `classpath` or `main` is provided without the other? (currently would end up going the `-jar` route)... and on a related note, it seems odd to be handling this decision while already in a method where the `jarPath` String has been passed in as a parameter. I wonder if we should be somehow checking for this main-class alternative somewhere earlier?